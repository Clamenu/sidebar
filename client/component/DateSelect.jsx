import React from 'react';
import styles from '../sidebar.css';

const DateSelect = ({ currentDate }) => (
  <div className={styles.dateContainer}>
    <img src='images/calendar.PNG'/>
    <input type='text' value={currentDate} readOnly className={styles.dateText}></input>
    <img src='images/downarrow.PNG' className={styles.dateArrowIcon}/>
  </div>
);

export default DateSelect;