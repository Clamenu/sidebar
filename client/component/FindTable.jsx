import React from 'react';
import styles from '../sidebar.css';

const FindTable = () => (
  <div className={styles.buttonContainer}>
    <a className={styles.button}>Find a Table</a>
  </div>
);

export default FindTable;
