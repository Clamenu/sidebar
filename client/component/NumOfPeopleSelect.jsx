import React from 'react';
import styles from '../sidebar.css';

const NumOfPeopleSelect = () => (
  <span className={styles.peopleContainer}>
    <img src='images/people.PNG' className={styles.icon}/>
    <select defaultValue='two' className={styles.select}>
      <option value='one'>1 person</option>
      <option value='two'>2 people</option>
      <option value='three'>3 people</option>
      <option value='four'>4 people</option>
      <option value='five'>5 people</option>
    </select>
    <img src='images/downarrow.PNG' className={styles.arrowIcon}/>
  </span>
);

export default NumOfPeopleSelect;