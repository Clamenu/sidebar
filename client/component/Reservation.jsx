import '../sidebar.css';
import React from 'react';

const Reservation = ({ reservationForm, currentDate }) => (
  <div>
    {reservationForm && (
      <label>
        <div className='reservation'>Make a Reservation</div><br/>
        <select>
          <option defaultValue='date'>{currentDate}</option>
        </select><br/>
        <select>
          
        </select>
        <select>
          <option value='one'>1 Person</option>
          <option defaultValue='two'>2 People</option>
          <option value='three'>3 People</option>
          <option value='four'>4 People</option>
          <option value='five'>5 People</option>
        </select><br/>
        <input type='submit' value='Find a Table' onClick={e => e.preventDefault()}></input>
      </label>
    )}
  </div>
);

export default Reservation;