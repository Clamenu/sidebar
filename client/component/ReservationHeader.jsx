import React from 'react';
import styles from '../sidebar.css';

const ReservationHeader = () => (
  <div className={styles.headerContainer}>
    <h3 className={styles.header}>
      <img src='images/header.PNG'/>
      <span className={styles.headerTitle} id='header-title'>Make a Reservation</span>
    </h3>
  </div>
);

export default ReservationHeader;