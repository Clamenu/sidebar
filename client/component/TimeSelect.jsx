import React from 'react';
import PropTypes from 'prop-types';
import styles from '../sidebar.css';

const TimeSelect = ({ timeslots }) => (
  <span className={styles.timeContainer}>
    <img src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/time.PNG" className={styles.icon} alt="#" />
    <select defaultValue="7:00 pm" className={styles.reservationSelect}>
      {timeslots.map((timeSlot) => {
        if (timeSlot !== '7:00 pm') {
          return (<option key={timeSlot} value={timeSlot}>{timeSlot}</option>);
        }
        return (<option selected key={timeSlot} value={timeSlot}>{timeSlot}</option>);
      })}
    </select>
    <img src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/downarrow.PNG" className={styles.arrowIcon} alt="#" />
  </span>
);

export default TimeSelect;

TimeSelect.propTypes = {
  timeslots: PropTypes.arrayOf(PropTypes.string).isRequired,
};
