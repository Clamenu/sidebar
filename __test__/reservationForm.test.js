import React from 'react';
import { shallow } from 'enzyme';
import Reservation from '../client/component/Reservation.jsx';
import ReservationHeader from '../client/component/ReservationHeader.jsx';
import FindTable from '../client/component/FindTable.jsx';
import NumOfPeopleSelect from '../client/component/NumOfPeopleSelect.jsx';
import DateSelect from '../client/component/DateSelect.jsx';
import TimeSelect from '../client/component/TimeSelect.jsx';

describe('Reservation Form React Component Testing', () => {
  describe('Reservation Header', () => {
    test('should contain text "Make a Reservation"', () => {
      const wrapper = shallow(<ReservationHeader/>);
      const text = wrapper.find('span').text();
      expect(text).toBe('Make a Reservation');
    });
    test('should match Reservation header snapshot', () => {
      const wrapper = shallow(<ReservationHeader/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Find a Table Button', () => {
    test('should contain value "Find a Table"', () => {
      const wrapper = shallow(<FindTable/>);
      const text = wrapper.find('a').text();
      expect(text).toBe('Find a Table');
    });
    test('should match Find A Table snapshot', () => {
      const wrapper = shallow(<FindTable/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Number of People select', () => {
    test('should have 5 options for number of people', () => {
      const wrapper = shallow(<NumOfPeopleSelect/>);
      const options = wrapper.find('select').children();
      expect(options).toHaveLength(5);
    });
    test('should match Num of People snapshots', () => {
      const wrapper = shallow(<NumOfPeopleSelect/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Date select', () => {
    test('should match current date that was given as a prop', () => {
      const wrapper = shallow(<DateSelect selectedDate='September 20, 2018'/>);
      const text = wrapper.find('input[type="text"]').prop('value');
      expect(text).toBe('September 20, 2018');
    });
    test('should not display calendar when displayCalendar is false', () => {
      const wrapper = shallow(<DateSelect displayCalendar={false}/>);
      const calendar = wrapper.find('div').children();
      expect(calendar).toHaveLength(3);
    });
    test('should display calendar when displayCalendar is true', () => {
      const wrapper = shallow(<DateSelect displayCalendar={true}/>);
      const calendar = wrapper.find('div').children();
      expect(calendar).toHaveLength(4);
    });
    test('should simulate a click to the text area to toggle calendar display', () => {
      const mockHandler = jest.fn();
      const wrapper = shallow(<DateSelect handleDateClick={mockHandler}/>);
      wrapper.find('input[type="text"]').simulate('click');
      expect(mockHandler.mock.calls.length).toBe(1);
    });
  });

  describe('Time select', () => {
    test('should have 3 options when given an array of 3 timeslots', () => {
      const wrapper = shallow(<TimeSelect timeslots={['7:00 pm', '7:30 pm', '8:00 pm']}/>);
      const select = wrapper.find('select').children();
      expect(select).toHaveLength(3);
    });
  });

  describe('Resevation', () => {
    test('should have 5 components in the reservation container field', () => {
      const wrapper = shallow(<Reservation hasReservation={true} />);
      const tree = wrapper.find('#reservation-container').children();
      expect(tree).toHaveLength(5); 
    });
  });
});
