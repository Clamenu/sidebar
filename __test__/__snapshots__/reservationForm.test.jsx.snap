// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation Form React Component Testing Find a Table Button should match Find A Table snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FindTable />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="button"
      >
        Find a Table
      </a>,
      "className": "buttonContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Find a Table",
        "className": "button",
      },
      "ref": null,
      "rendered": "Find a Table",
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="button"
        >
          Find a Table
        </a>,
        "className": "buttonContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find a Table",
          "className": "button",
        },
        "ref": null,
        "rendered": "Find a Table",
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Reservation Form React Component Testing Number of People select should match Num of People snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NumOfPeopleSelect />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="#"
          className="peopleIcon"
          src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/people.PNG"
        />,
        <select
          className="reservationSelect"
          defaultValue="two"
        >
          <option
            value="one"
          >
            1 person
          </option>
          <option
            value="two"
          >
            2 people
          </option>
          <option
            value="three"
          >
            3 people
          </option>
          <option
            value="four"
          >
            4 people
          </option>
          <option
            value="five"
          >
            5 people
          </option>
        </select>,
        <img
          alt="#"
          className="arrowIcon"
          src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/downarrow.PNG"
        />,
      ],
      "className": "peopleContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "#",
          "className": "peopleIcon",
          "src": "https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/people.PNG",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="one"
            >
              1 person
            </option>,
            <option
              value="two"
            >
              2 people
            </option>,
            <option
              value="three"
            >
              3 people
            </option>,
            <option
              value="four"
            >
              4 people
            </option>,
            <option
              value="five"
            >
              5 people
            </option>,
          ],
          "className": "reservationSelect",
          "defaultValue": "two",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1 person",
              "value": "one",
            },
            "ref": null,
            "rendered": "1 person",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2 people",
              "value": "two",
            },
            "ref": null,
            "rendered": "2 people",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "3 people",
              "value": "three",
            },
            "ref": null,
            "rendered": "3 people",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4 people",
              "value": "four",
            },
            "ref": null,
            "rendered": "4 people",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5 people",
              "value": "five",
            },
            "ref": null,
            "rendered": "5 people",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "#",
          "className": "arrowIcon",
          "src": "https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/downarrow.PNG",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="#"
            className="peopleIcon"
            src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/people.PNG"
          />,
          <select
            className="reservationSelect"
            defaultValue="two"
          >
            <option
              value="one"
            >
              1 person
            </option>
            <option
              value="two"
            >
              2 people
            </option>
            <option
              value="three"
            >
              3 people
            </option>
            <option
              value="four"
            >
              4 people
            </option>
            <option
              value="five"
            >
              5 people
            </option>
          </select>,
          <img
            alt="#"
            className="arrowIcon"
            src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/downarrow.PNG"
          />,
        ],
        "className": "peopleContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "#",
            "className": "peopleIcon",
            "src": "https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/people.PNG",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="one"
              >
                1 person
              </option>,
              <option
                value="two"
              >
                2 people
              </option>,
              <option
                value="three"
              >
                3 people
              </option>,
              <option
                value="four"
              >
                4 people
              </option>,
              <option
                value="five"
              >
                5 people
              </option>,
            ],
            "className": "reservationSelect",
            "defaultValue": "two",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1 person",
                "value": "one",
              },
              "ref": null,
              "rendered": "1 person",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2 people",
                "value": "two",
              },
              "ref": null,
              "rendered": "2 people",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3 people",
                "value": "three",
              },
              "ref": null,
              "rendered": "3 people",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4 people",
                "value": "four",
              },
              "ref": null,
              "rendered": "4 people",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5 people",
                "value": "five",
              },
              "ref": null,
              "rendered": "5 people",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "#",
            "className": "arrowIcon",
            "src": "https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/downarrow.PNG",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Reservation Form React Component Testing Reservation Header should match Reservation header snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReservationHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3
        className="header"
      >
        <img
          alt="#"
          className="headerIcon"
          src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/header.PNG"
        />
        <span
          className="headerTitle"
          id="header-title"
        >
          Make a Reservation
        </span>
      </h3>,
      "className": "headerContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="#"
            className="headerIcon"
            src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/header.PNG"
          />,
          <span
            className="headerTitle"
            id="header-title"
          >
            Make a Reservation
          </span>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "#",
            "className": "headerIcon",
            "src": "https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/header.PNG",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Make a Reservation",
            "className": "headerTitle",
            "id": "header-title",
          },
          "ref": null,
          "rendered": "Make a Reservation",
          "type": "span",
        },
      ],
      "type": "h3",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3
          className="header"
        >
          <img
            alt="#"
            className="headerIcon"
            src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/header.PNG"
          />
          <span
            className="headerTitle"
            id="header-title"
          >
            Make a Reservation
          </span>
        </h3>,
        "className": "headerContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="#"
              className="headerIcon"
              src="https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/header.PNG"
            />,
            <span
              className="headerTitle"
              id="header-title"
            >
              Make a Reservation
            </span>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "#",
              "className": "headerIcon",
              "src": "https://s3-us-west-1.amazonaws.com/yelpreactorsidebaricons/header.PNG",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Make a Reservation",
              "className": "headerTitle",
              "id": "header-title",
            },
            "ref": null,
            "rendered": "Make a Reservation",
            "type": "span",
          },
        ],
        "type": "h3",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
